consensus/consensus_v2.go

```diff
@@ -762,8 +765,43 @@ func (consensus *Consensus) onCommit(msg *msg_pb.Message) {
 		Int64("numReceivedSoFar", consensus.Decider.SignersCount(quorum.Commit)).
 		Logger()
 	logger.Info().Msg("[OnCommit] Received new commit message")
+
+	// proceed only when the message is not received before
+	if signed := consensus.Decider.ReadBallot(
```

Comment:
Moving the check lower here will incur lots of unnecessary checks above.

Reply:
ya, I thought about that but there is information that i need and i didn't want to spend time on more thinking of how to organize both.